lib/DocumentManager.php

```diff
@@ -27,11 +39,62 @@ class DocumentManager implements DocumentManagerInterface
     private $optionsResolvers = [];
 
     /**
-     * @param EventDispatcherInterface $eventDispatcher
+     * @var DocumentInspectorFactory
      */
-    public function __construct(EventDispatcherInterface $eventDispatcher)
-    {
+    private $inspectorFactory;
+
+    /**
+     * @var NodeManager
+     */
+    private $nodeManager;
+
+    /**
+     * @var DocumentRegistry
+     */
+    private $registry;
+
+    /**
+     * @var ProxyFactory
+     */
+    private $proxyFactory;
+
+    /**
+     * @var MetadataFactoryInterface
+     */
+    private $metadataFactory;
+
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    public function __construct(
+        SessionInterface $session,
+        EventDispatcherInterface $eventDispatcher,
+        MetadataFactoryInterface $metadataFactory,
+        LazyLoadingGhostFactory $lazyProxyFactory,
+        DocumentInspectorFactoryInterface $inspectorFactory,
+        $defaultLocale
+    ) {
+        // the event dispatcher should be unique to this document manager
+        // instance.
         $this->eventDispatcher = $eventDispatcher;
+
+        // the inspector factory provides a way for users to instantiate
+        // their own document inspectors.
+        $this->inspectorFactory = $inspectorFactory;
+
+        // the metadata factory is currently intended to be the same for all
+        // document manager instances.
+        $this->metadataFactory = $metadataFactory;
+
+        // the PHPCR session SHOULD be unique to all document managers.
+        $this->session = $session;
+
+        // instantiate other objects scoped to this document manager.
+        $this->nodeManager = new NodeManager($session);
+        $this->proxyFactory = new ProxyFactory($this, $lazyProxyFactory, $metadataFactory);
+        $this->registry = new DocumentRegistry($defaultLocale);
```

Comment:
Why not also do so with the `DocumentInspector`? Don't say it would be better, but at least more consistent. However I am not really sure about that at the moment.